app/src/main/java/com/graddex/graddex/PokemonDetailsFragment.kt

```diff
@@ -0,0 +1,96 @@
+package com.graddex.graddex
+
+import android.os.Bundle
+import android.util.Log
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.os.bundleOf
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.ViewModelProvider
+import coil.load
+import com.graddex.graddex.R.string.*
+import com.graddex.graddex.models.PokemonDetailsViewModel
+import com.graddex.graddex.databinding.FragmentPokemonDetailsBinding
+import java.util.*
+
+private const val pokemonKey: String = "pokemonKey"
+
+class PokemonDetailsFragment(args: Bundle) : Fragment() {
+
+    constructor(id: String) : this(bundleOf(pokemonKey to id))
+
+    private val pokemonId: String = args.getString(pokemonKey, "")
+
+    private lateinit var binding: FragmentPokemonDetailsBinding
+    private lateinit var viewModel: PokemonDetailsViewModel
+
+    override fun onCreateView(
+            inflater: LayoutInflater, container: ViewGroup?,
+            savedInstanceState: Bundle?
+    ): View? {
+
+        Log.d(tag, "PokemonID: $pokemonId")
+
+        // Inflate the Pokemon Details fragment layout
+        binding = DataBindingUtil.inflate<FragmentPokemonDetailsBinding>(
+            inflater, R.layout.fragment_pokemon_details, container, false
+        )
+
+        // Call the Pokemon Details view model for specified Pokemon ID
+        viewModel = ViewModelProvider(this)
+            .get(PokemonDetailsViewModel::class.java)
+
+        binding.statusText.text = getString(loading)
+        viewModel.syncPokemonDetails(pokemonId)
+
+        return binding.root
+
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        // Update UI with details
+        viewModel.pokemonSpriteFront.observe(viewLifecycleOwner,
+            { pokemonSpriteFront ->
+                binding.pokemonFrontImage.load(pokemonSpriteFront)
+                binding.statusText.visibility = View.GONE
+            })
```

Comment:
This is a neat feature with kotlin:

when you have blocks like:
```
{ pokemonSpriteFront ->
    binding.pokemonFrontImage.load(pokemonSpriteFront)
    binding.statusText.visibility = View.GONE
}
```
as the last parameter, you can actually move it out of the parameter list like this (kinda like the `listener` for the `PokemonRecyclerAdapter` https://github.com/GradDex/GradDex/pull/15/files#diff-c42c98b09035ebb40367f76a1f0dae649f2eeb76a1f732cd7c143503f4bfe666R35):
```suggestion
        viewModel.pokemonSpriteFront.observe(viewLifecycleOwner) { pokemonSpriteFront ->
            binding.pokemonFrontImage.load(pokemonSpriteFront)
            binding.statusText.visibility = View.GONE
        }
```

It makes it a bit easier to read because of the indentation

Reply:
done :)